### PR TITLE
Update NFRs and Glossary in DeveloperGuide.md

### LINKED ISSUE #33: Update NFRs and Glossary in DeveloperGuide.md

### PR DESCRIPTION
**Pull request by prave1n**
Fixes #33

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -341,16 +341,50 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+* **DBMS (Database Management System)**: A software system that enables users to define, create, maintain and control access to a database. It provides an organised way of managing, storing, and retrieving vast amounts of data.
+* **Delivery Information**: Details about a specific delivery, including date, supplier, products, and status.
+* **Human-Editable File**: A plain text file that can be easily opened, read, and modified by users, typically in formats such as JSON, CSV, or TXT, without needing specialised software.
+* **JAR File**: A Java ARchive file, which is a package file format that aggregates many Java class files and associated resources (text, images, etc.) into one file for distribution.
+* **Mainstream OS**: Windows, Linux, Unix, macOS
+* **OOP (Object-Oriented Programming)**: A programming paradigm based on the concept of objects, which can contain data and methods.
+* **Supplier Contact**: A record containing information about a supplier, including name, contact details, and associated products.
+  
+*{More to be added}*
```
Comment (LinkesV) [inline docs/DeveloperGuide.md:387]: Response Time 
Unit and Integration Testing
 breadth-first incremental manner

I think these 3 terms should be defined here

**Review by LinkesV — COMMENTED** [review]
Other than the missing 3 terms this looks good to me

**Review by LinkesV — APPROVED** [review]
LGTM!!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (prave1n) [reply docs/DeveloperGuide.md:392]: Yes Linkes, I agree that those terms must be defined in the Glossary. Thank you for your inputs, I have now made the necessary changes in my latest commit in this branch

**Pull request merged**